scipy/sparse/linalg/isolve/tests/test_minres.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from numpy.linalg import norm
+from numpy.testing import (assert_equal, assert_array_equal, assert_allclose,
+                           assert_, assert_raises, run_module_suite, dec)
+from scipy.sparse import diags, eye, lil_matrix
+from scipy.sparse.linalg.isolve import minres
+
+# A Random 10 x 10 sparse matrix
+np.random.seed(1234)
+matrix = diags(np.random.rand(10), format='csr')
+# A Random Vector of length 10
+np.random.seed(1234)
+vector = np.random.rand(10)
+
+
+def assert_normclose(a, b, tol=1e-5):
+    residual = norm(a - b)
+    tolerance = tol * norm(b)
+    msg = "residual (%g) not smaller than tolerance %g" % (residual, tolerance)
+    assert_(residual < tolerance, msg=msg)
+
+
+def test_singular():
+    A, b = matrix.copy(), vector.copy()
+    A[(0, 0)], b[0] = 0, 0
+    xp, info = minres(A, b)
+
+    assert_equal(info, 0)
+    assert_normclose(A.dot(xp), b)
+
+
+@dec.skipif(True, "Skip Until gh #6843 is fixed")
+def test_gh_6843():
+    '''check if x0 is being used by tracing iterates'''
+    A, b = matrix, vector
+    # Random x0 to feed minres
+    np.random.seed(1234)
+    x0 = np.random.rand(10)
+    trace = []
+
+    def trace_iterates(xk):
+        trace.append(xk)
+    minres(A, b, x0=x0, callback=trace_iterates)
+    trace_with_x0 = trace
+
+    trace = []
+    minres(A, b, callback=trace_iterates)
+    assert_raises(AssertionError, assert_array_equal, trace_with_x0, trace)
+
+
+def test_shift():
+    A, b = matrix, vector
+
+    # A Random integer for shift
+    np.random.seed(1234)
+    shift = np.random.randint(1)
+
+    shifted_A = A - shift * eye(10)
+    x1, info1 = minres(A, b, shift=shift)
+    x2, info2 = minres(shifted_A, b)
+    assert_equal(info1, 0)
+    assert_allclose(x1, x2)
+
+
+def test_asymmetric_fail():
+    '''Asymmetric matrix should raise `ValueError` when check=True'''
+    with assert_raises(ValueError):
```

Comment:
Put only the `minres` call inside this `with assert` --- the test shouldn't pass if some other part of the code raises valueerror